### PR TITLE
chore: move fetchAppOpportunities into react-queries

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Refactors the opportunities react-query to adopt the existing pattern in our project, `createQueryKeys`.

This task required moving of files to resolve circular dependencies, but no functional change should be present with this PR.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Follow-up to #6409

Low-ish risk as this is mostly moving of files.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

Check the following are still working:
* Staking opportunities fetching
* Total balance on wallet page
* Defi balance on wallet page
* ERC20 allowance checks (in trade etc)
* Lending lock
* LP available pools
* LP positions
* LP add liquidity
* LP remove liquidity
* LP status on transaction row
* Trades halted check for thorchain

### CODE REVIEW

**Comment by woodenfurniture** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/6449?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6449** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6449?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6409** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6409?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @woodenfurniture and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import type { AssetId } from '@shapeshiftoss/caip'
-import { reactQueries } from 'react-queries'
+import { thornode } from 'react-queries/queries/thornode'
```
Comment (woodenfurniture) [inline src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/index.ts:2]: This is the only change which could be problematic. Required to break dependency cycle. Seems to work fine

**Review by 0xApotheosis — APPROVED** [review]
Spot checked this branch vs develop, no delta outside of expected noise.

Screenshot 2024-03-14 at 3 33 07 PM
Screenshot 2024-03-14 at 3 33 34 PM
Screenshot 2024-03-14 at 3 33 52 PM
Screenshot 2024-03-14 at 3 34 54 PM
Screenshot 2024-03-14 at 3 35 02 PM

File: src/components/StakingVaults/hooks/useFetchOpportunities.tsx [hunk 2]
```diff
@@ -43,51 +29,15 @@ export const useFetchOpportunities = () => {
...
-      return null
-    },
-    // once the portfolio is loaded, fetch opportunities data
-    enabled: portfolioLoadingStatus !== 'loading' && requestedAccountIds.length > 0,
-    refetchInterval: false,
-    staleTime: Infinity,
-    gcTime: Infinity,
-  })
+  const { isLoading } = useQuery(
+    reactQueries.opportunities.all(
```
Comment (gomesalexandre) [inline src/components/StakingVaults/hooks/useFetchOpportunities.tsx:33]: Fitting the pattern of `react-query` would probably be using `useQueries` here, though I get it that's not necessarily straightforward to make this work with the opportunities abstraction!

File: src/react-queries/index.ts [hunk 3]
```diff
@@ -1,265 +1,17 @@
...
-  getErc20Allowance,
-  getFees,
-} from 'lib/utils/evm'
-import type { ThorchainBlock, ThorchainMimir } from 'lib/utils/thorchain/lending/types'
-import type {
-  MidgardPoolStats,
-  MidgardSwapHistoryResponse,
-  MidgardTvlHistoryResponse,
-} from 'lib/utils/thorchain/lp/types'
+import { mergeQueryKeys } from '@lukemorales/query-key-factory'
```
Comment (gomesalexandre) [inline src/react-queries/index.ts:1]: 👀

File: src/react-queries/index.ts [hunk 4]
```diff
@@ -1,265 +1,17 @@
...
-
-export const reactQueries = mergeQueryKeys(common, mutations, midgard, thornode, thorchainLp)
+import { common } from './queries/common'
+import { midgard } from './queries/midgard'
+import { mutations } from './queries/mutations'
+import { opportunities } from './queries/opportunities'
+import { thornode } from './queries/thornode'
+
+export const reactQueries = mergeQueryKeys(
+  common,
```
Comment (gomesalexandre) [inline src/react-queries/index.ts:11]: Ty for the cleanup moving these to their own module! 🙏🏽

File: src/state/slices/opportunitiesSlice/thunks.ts [hunk 5]
```diff
@@ -1,14 +1,15 @@
...
-import { store } from 'state/store'
+import type { AppDispatch } from 'state/store'
 
 import { foxEthStakingIds } from '../opportunitiesSlice/constants'
 import { CHAIN_ID_TO_SUPPORTED_DEFI_OPPORTUNITIES } from './mappings'
 import { opportunitiesApi } from './opportunitiesApiSlice'
 import { DefiProvider, DefiType } from './types'
 
 export const fetchAllLpOpportunitiesMetadataByChainId = async (
+  dispatch: AppDispatch,
```
Comment (gomesalexandre) [inline src/state/slices/opportunitiesSlice/thunks.ts:12]: Was this passed around to avoid circular deps? `store.dispatch` effectively is the same

**Review by gomesalexandre — COMMENTED** [review]
Conceptually looks sane to me pending sanity test!

**Review by gomesalexandre — APPROVED** [review]
Tested locally, didn't find any regressions against develop except for the same non-deterministic issues as before!

File: src/state/slices/opportunitiesSlice/thunks.ts (hunk 5 above)
Comment (woodenfurniture) [reply src/state/slices/opportunitiesSlice/thunks.ts:12]: correct yeah, we cannot consume `import { store } from 'state/store'` from these due to circ deps.

**Pull request merged**